core/src/main/java/com/linecorp/armeria/server/file/AbstractHttpFile.java

```diff
@@ -197,7 +200,15 @@ public final CompletableFuture<HttpResponse> read(Executor fileReadExecutor, Byt
         requireNonNull(alloc, "alloc");
 
         return readAttributes(fileReadExecutor)
-                .thenApply(attrs -> read(fileReadExecutor, alloc, attrs))
+                .thenApply(attrs -> {
+                    final HttpResponse res = read(fileReadExecutor, alloc, attrs);
+                    if (res != null) {
+                        return res;
+                    }
+
+                    logger.warn("{} is not found. attrs: {}", pathOrUri(), attrs);
```

Comment:
How about changing `warn` to `debug` or `trace`? I don't see failing to find a file is a warning.
It commonly happens if a user types a wrong URL or a client requests an old file that was deleted before.

Reply:
The invalid request shouldn't be seen as a warning, I'll fix that for easy debugging.